### PR TITLE
Data page query strings

### LINKED ISSUE #420: build new Explore the Data landing page
Please build a new landing page for our "Explore the Data" tab. We would like this to include cards of our different interactive data tools in the style of [Reuters' page for interactives.](http://graphics.reuters.com/), to include these cards: 
- Texas shootings by law enforcement 
- Texas shootings of law enforcement 
- Custodial Deaths in Texas; 
- COVID-19 deaths in Texas incarceration facilities; 
- COVID-19 deaths of Texas officers (still being built); 
- Pre-trial deaths in Texas jails

### PR DESCRIPTION
**Pull request by LayaTaal**
Updates existing OIS and Custodial Death datasets to use dynamic routing rather than all living on a single page. This will allow us to link directly to each data set individually from the new interactive data page.

Since this change creates a new url based on the data set name, it also [updates the naming convention](https://github.com/texas-justice-initiative/website-nextjs/blob/ccf5366cc611944e12948c6fc13d034a46a20176/data/datasets.js#L10) in the data set schema to use dashes rather than camel case. 

Closes #420

### CODE REVIEW

File: pages/data.js [hunk 1]
```diff
@@ -198,20 +220,30 @@ export default class Explore extends React.Component {
...
+                    }}
                   >
-                    <span className="btn--chart-toggle--icon">
-                      <img src={datasets[datasetName].icon} alt={datasets[datasetName].name} />
-                    </span>
-                    <span className="btn--chart-toggle--text">{datasets[datasetName].name}</span>
-                  </ChangeChartButton>
+                    <a
+                      onClick={() => this.fetchData(datasetName)}
+                      onKeyDown={() => this.fetchData(datasetName)}
```
Comment (nholden) [inline pages/data.js:232]: Do we need these `onClick` and `onKeyDown` handlers now that we're linking to a new path?

File: pages/data.js [hunk 2]
```diff
@@ -276,20 +308,30 @@ export default class Explore extends React.Component {
...
+                  }}
                 >
-                  <span className="btn--chart-toggle--icon">
-                    <img src={datasets[datasetName].icon} alt={datasets[datasetName].name} />
-                  </span>
-                  <span className="btn--chart-toggle--text">{datasets[datasetName].name}</span>
-                </ChangeChartButton>
+                  <a
+                    onClick={() => this.fetchData(datasetName)}
+                    onKeyDown={() => this.fetchData(datasetName)}
```
Comment (nholden) [inline pages/data.js:320]: Same q as above.

File: pages/data.js [hunk 3]
```diff
@@ -30,9 +32,25 @@ export default class Explore extends React.Component {
...
+  static getInitialProps({ query }) {
+    return { query };
+  }
+
+  /**
+   * Once component mounts, check for query string and load
+   * dataset if it exists, else load the first dataset.
+   */
   componentDidMount() {
+    const { query } = this.props;
```
Comment (nholden) [inline pages/data.js:44]: Curious: what happened when you tried to access the query params through `next/router`? Error message, nothing?

File: package-lock.json [hunk 4]
```diff
@@ -6954,7 +6954,8 @@
         },
         "ansi-regex": {
           "version": "2.1.1",
-          "bundled": true
+          "bundled": true,
```
Comment (nholden) [inline package-lock.json:6957]: What do these changes to `package-lock.json` do?

File: pages/datasets/[dataset].js [hunk 5]
```diff
@@ -0,0 +1,551 @@
...
+      data: {},
+      filters: {},
+    };
+
+    this.updateFilters = this.updateFilters.bind(this);
+    this.updateFilterGroup = this.updateFilterGroup.bind(this);
+  }
+
+  /**
+   * Once component mounts, check for query string and load
```
Comment (nholden) [inline pages/datasets/[dataset].js:36]: Should we update this comment now that we're not using query strings?

**Review by nholden — APPROVED** [review]
Looks great, @LayaTaal! 😎 

I noticed that the URLs automatically get lower cased when I visit the datasets in the deploy preview. That makes me think that it would be worth updating the URLs to `kebab-case` instead of `camelCase`, like you suggested in Slack.

For posterity, I also think it would be worth updating the PR description to reflect how this works now.

File: pages/datasets/[dataset].js (hunk 5 above)
Comment (LayaTaal) [reply pages/datasets/[dataset].js]: Updated and also removed the no longer necessary `componentDidUpdate` lifecycle.

**Comment by LayaTaal** [comment]
@nholden updated to address your feedback. Dataset slugs should all now be dashes instead of camel cased. If you could take a spin through the preview that would be great!

**Review by nholden — APPROVED** [review]
Deploy preview looks 💯  @LayaTaal!

Hoping that once you merge in `main` that the data license modal I worked on in https://github.com/texas-justice-initiative/website-nextjs/pull/503 will just work. 🤞 😅

**Pull request merged**